lib/agent.js

```diff
@@ -135,14 +135,11 @@ Agent.prototype.destroy = function () {
   this._instrumentation.stop();
 
   if (this._otelMeterProvider) {
-    // Dev Note: It is unfortunate that `.destroy()` can return while this
-    // shutdown work might still be ongoing. See
-    // https://github.com/elastic/apm-agent-nodejs/issues/3222 to fix that.
-    this._otelMeterProvider
-      .shutdown({ timeoutMillis: 1000 })
-      .catch((reason) => {
-        this.logger.warn('failed to shutdown OTel MeterProvider:', reason);
-      });
+    try {
+      await this._otelMeterProvider.shutdown({ timeoutMillis: 1000 });
+    } catch (reason) {
+      this.logger.warn('failed to shutdown OTel MeterProvider:', reason);
+    }
     this._otelMetricsProvider = null;
```

Comment:
I just noticed this var is misspelled.
```suggestion
    this._otelMeterProvider = null;
```